Realm/RLMRealm.mm

```diff
@@ -390,9 +381,9 @@ + (instancetype)realmWithPath:(NSString *)path
         return RLMAutorelease(realm);
     }
 
-    key = key ?: keyForPath(path);
-    realm = [[RLMRealm alloc] initWithPath:path key:key readOnly:readonly inMemory:inMemory dynamic:dynamic error:outError];
-    if (outError && *outError) {
+    NSData *key = configuration.encryptionKey ?: keyForPath(path);
```

Comment:
Does this key ever get validated? It would be worth considering moving that validation to RLMConfiguration.